ql/src/java/org/apache/hadoop/hive/ql/DriverContext.java

```diff
@@ -214,6 +219,7 @@ public long getCompactorTxnId() {
 
   public void setCompactorTxnId(long compactorTxnId) {
     this.compactorTxnId = compactorTxnId;
+    conf.setLong(ValidTxnList.COMPACTOR_VALID_TXNS_ID_KEY, compactorTxnId);
```

Comment:
if not being used may be move this out of this patch.

Reply:
We currently maintain the validWriteIds in DriverContext.